UI: Breadcrumb home icon + left-nav animation/spacings

johndmulhausen opened the pull request:
This PR adds a bunch of CSS tweaks and also a couple of features:

- Breadcrumb: render a Home icon as the first item via the Hugo partial (no JS injection). 
- Left nav: adds smooth expand/collapse animation, rotating caret, fixes caret clipping, and normalizes spacing before/after child groups.

Partially inspired by me wanting to try GPT-5 :)

johndmulhausen committed:
Import breadcrumb CSS styling and house icon from css-improvements branch

johndmulhausen committed:
Import scrolling synchronization logic from css-improvements branch

- Added TOCScrollSync JavaScript class to track scroll position
- Highlights active TOC links as user scrolls through content
- Added CSS transitions and active state styling for TOC links
- Fixed SCSS syntax error for breakpoint calculation

johndmulhausen committed:
Import font family changes from css-improvements branch

- Added -font variable using system fonts instead of Source Sans Pro
- Removed hardcoded font-family declarations from body and headings
- Updated p, td, div, li, .lead to use -font with proper sizing
- Changed all headings to use bold font weight
- Updated heading sizes: h4 (16px), h5 (14px), h6 (12px)
- Removed Source Serif 4 font from headings
- Updated sidebar nav font size to 13px
- Fixed empty .gs-title ruleset

johndmulhausen committed:
Increase main reading font size from 14px to 16px for better readability

johndmulhausen committed:
Fix right-nav (TableOfContents) font size inheritance issue

- Added explicit 13px font-size to TableOfContents links and list items
- Prevents inheritance of 16px from general li styling rule

johndmulhausen committed:
Align left and right navigation fonts

- Both sidebars now use $base-font (system fonts) instead of Source Sans Pro
- Aligned font sizes to 13px for both navigations
- Ensures consistent typography across all navigation elements

johndmulhausen committed:
Increase left navigation font size to 14px

- Left sidebar navigation now uses 14px (up from 13px)
- Section titles also updated to 14px for consistency
- Right navigation (TableOfContents) remains at 13px
- Better visual hierarchy between primary and secondary navigation

johndmulhausen committed:
Import hover transitions from css-improvements branch

- Added card hover animations: translateY(-3px) with drop-shadow effect
- Added navbar link hover transform: translateY(-2px) for subtle lift
- Added sidebar navigation hover transitions: translateX(2px) slide effect
- Added ul.foldable transition for smooth animations
- All transitions use 0.2s ease-in-out timing for consistent feel
- Creates subtle sense of responsiveness throughout the UI

johndmulhausen committed:
Enhance navigation spacing and remove redundant breadcrumb JS

- Added consistent margin-bottom spacing for parent items and child lists in foldable navigation
- Applied the same spacing rule to all nested levels for improved layout
- Removed JavaScript injection for home icon in breadcrumb; now rendered via the breadcrumb partial

johndmulhausen committed:
Refine foldable navigation styles for improved spacing and transitions

- Adjusted margin-bottom for sidebar navigation items to ensure consistent spacing
- Enhanced padding for chevrons to prevent clipping and improve layout
- Implemented smooth expand/collapse transitions for nested lists
- Removed redundant margins after last child items for cleaner appearance

johndmulhausen committed:
Adjust foldable navigation transform for improved optical alignment

- Modified vertical translation for navigation elements to enhance visual consistency
- Ensured better alignment of checked states in foldable navigation

johndmulhausen committed:
Update breadcrumb partial to exclude home link from breadcrumb items

- Added conditional check to prevent rendering the home link in the breadcrumb trail
- Enhances breadcrumb clarity by omitting the home item when displaying ancestors

mdlinville commented on assets/scss/_variables_project.scss:
```diff
@@ -117,34 +118,56 @@ body {
+    font-size: 16px !important;
+    font-weight: $font-weight-bold !important;
+}
+h5, .td-content h5 {
```
This is now the same as the base font, which means that H6 is smaller than the base font. WDYT?

johndmulhausen replied:
Good catch; fixed.

mdlinville commented on assets/scss/_variables_project.scss:
```diff
@@ -6,6 +6,8 @@ Add styles or override variables from the theme here.
 
 @import url('https://fonts.googleapis.com/css2?family=Source+Serif+4:ital,opsz,wght@0,8..60,200..900;1,8..60,200..900&display=swap');
 
+$base-font: system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
```
This is quite a few font face changes. Are we intentionally going away from Source Sans Pro?

johndmulhausen replied:
Oh totally; looks way better and is the font of choice on the Weave docs.

mdlinville commented on assets/scss/_variables_project.scss:
```diff
@@ -488,20 +560,31 @@ ol.breadcrumb li:not(:first-child):before {
         margin: 0px !important;
         padding: 8px !important;
         font-weight: 500 !important;
+        font-size: 13px !important;
```
I wonder if this should also be 12 to align with the breadcrumbs, which are also a navigation element. Also not sure you need line 564 and 569 both. It would be bad if they went out of sync.

johndmulhausen replied:
Good calls on both things. Fixed.

mdlinville commented on assets/scss/_variables_project.scss:
```diff
@@ -488,20 +560,31 @@ ol.breadcrumb li:not(:first-child):before {
         li {
             padding: 0px !important; 
             line-height: 14px;
+            font-size: 13px !important;
```
Same feedback here about the font size. Probably all of the links in the navigation are in `<ul><li>` hierarchy.

johndmulhausen replied:
Right, removed this definition; everything should be styled by just the font-size definition in `a` now

mdlinville commented on assets/scss/_variables_project.scss:
```diff
@@ -1083,18 +1173,21 @@ div.td-page-meta__lastmod {
-    font-family: "Source Sans Pro", sans-serif;
+    font-family: $base-font;
     padding-left: 15px !important;
+    font-size: 14px !important; // Slightly larger than right nav
```
This one is 14 as opposed to 13 above, and these are still navigation `<ul><li>`, right?

johndmulhausen replied:
Right-nav is now 12, so I think having the left-nav be 14 makes sense; page titles are shorter than headings generally and we frequently shorten them even more with the `linkTitle` front-matter for pure "scan-ability." This one I'm calling good. However, we'll address font-weight below. :)

mdlinville reviewed (changes requested):
Feedback from skimming the CSS changes. Next I will spot check some pages in case there is more feedback there.

mdlinville reviewed (changes requested):
<img width="499" height="47" alt="Screenshot 2025-08-08 at 3 32 54 PM" src="https://github.com/user-attachments/assets/958833a5-5a03-444c-9575-bc605ed9e28e" />

Maybe my eyes are playing tricks, but the terminal breadcrumb looks less bold than the others and the black on gray is harder to read than the teal with no background color.

<img width="519" height="363" alt="Screenshot 2025-08-08 at 3 34 46 PM" src="https://github.com/user-attachments/assets/9ee5a076-4ccf-4d07-8204-a580068ba48a" />

To my eyes, these headings look way too heavy.

<img width="690" height="390" alt="Screenshot 2025-08-08 at 3 35 39 PM" src="https://github.com/user-attachments/assets/7f9b89d8-f6cd-4f47-85a2-e8b2e562c221" />

These titles also look heavy and large.

<img width="253" height="131" alt="Screenshot 2025-08-08 at 3 36 48 PM" src="https://github.com/user-attachments/assets/968f9a2f-1d3c-4788-ba74-99563a0f848d" />

Code font in paragraph text looks quite large to me.

mdlinville commented:
One more: In dark mode, the left nav looks too light / not bold enough. I think it is an optical illusion, but it may point to a need to make the navigation a little heavier in both modes.
<img width="573" height="524" alt="Screenshot 2025-08-08 at 3 39 47 PM" src="https://github.com/user-attachments/assets/7e3b7a86-e983-49c7-99d1-0fa4f6ed52e6" />

johndmulhausen commented:
> Maybe my eyes are playing tricks, but the terminal breadcrumb looks less bold than the others and the black on gray is harder to read than the teal with no background color.
> 
> To my eyes, these headings look way too heavy.
> 
> These titles also look heavy and large.
> 
> Code font in paragraph text looks quite large to me.

Good call on the breadcrumbs. I am matching the font weight on Docusaurus re: headings and sizes so I'm going to leave those as-is, I think they look great.  Re: inline code, though, I think what was making them look off was the background color on the pills was nigh transparent. I think properly framed by a visible pill it makes a lot more sense now.

> One more: In dark mode, the left nav looks too light / not bold enough. I think it is an optical illusion, but it may point to a need to make the navigation a little heavier in both modes. 

Bolded up the left-nav a bit. 

Responded to the other bits inline.

johndmulhausen committed:
Update heading sizes and background colors in SCSS variables

- Adjusted font sizes for h4, h5, and h6 headings for better hierarchy
- Updated background color values for hover and active states in navigation elements
- Introduced inline code styles for improved visibility and consistency
- Refined breadcrumb background color for active state
- Ensured consistent font weight for headings across the project

johndmulhausen committed:
Limit content width in SCSS for improved readability

- Added max-width of 75ch to .td-content for optimal text layout
- Set margin to 0 to left-align content, enhancing overall presentation

johndmulhausen committed:
Add mobile-specific styles for improved layout and spacing

mdlinville commented on assets/scss/_variables_project.scss:
```diff
@@ -488,20 +560,31 @@ ol.breadcrumb li:not(:first-child):before {
 #TableOfContents {
-    
+    font-family: $base-font;
     font-size: 13px !important;
```
Is this one still redundant?

johndmulhausen replied:
It is, zapped

johndmulhausen committed:
Enhance typography in sidebar navigation and language selector

- Updated font sizes and weights for sidebar section titles and language selector links to improve visual consistency
- Applied !important declarations to ensure styles are prioritized
- Added padding and margin adjustments for better layout in the language selector

johndmulhausen committed:
Remove width constraint on main content in SCSS for improved layout flexibility

johndmulhausen committed:
Update margin-bottom for navigation links in SCSS to enhance spacing consistency

johndmulhausen committed:
Remove font-family declaration from Table of Contents component in SCSS to streamline styling

johndmulhausen committed:
Merge branch 'main' into feature/breadcrumb-home-and-leftnav-polish

johndmulhausen commented:
I meant to just close this; as a thought experiment it was fun but given that we're moving to Mintlify it's a pointless exercise. After merging, instantly reverted. This can stay dead.
